### PR TITLE
Fix NRE crashes on OS X

### PR DESCRIPTION
**Pull request by rainersigwald**
This should fix crashes of this form:
```
Unhandled Exception: System.TypeInitializationException: The type initializer for 'Microsoft.Build.CommandLine.MSBuildApp' threw an exception. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Build.Shared.FileUtilities.get_ExecutingAssemblyPath()
   at Microsoft.Build.CommandLine.MSBuildApp..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.Build.CommandLine.MSBuildApp.Main(String[] args)
```

They occurred because modules aren't supported on the OS X runtime currently. Fortunately we could avoid using modules in most cases.

### CODE REVIEW

File: src/XMakeBuildEngine/Construction/ProjectRootElement.cs [hunk 1]
```diff
@@ -1956,7 +1956,9 @@ private XmlDocumentWithLocation LoadDocument(string fullPath)
                     XmlReaderSettings xmlReaderSettings = new XmlReaderSettings();
                     xmlReaderSettings.CloseInput = true; // close the stream when disposing the reader
                     xmlReaderSettings.DtdProcessing = DtdProcessing.Ignore;
-                    using (XmlReader xr = XmlReader.Create(File.OpenRead(fullPath), xmlReaderSettings))
+
+                    var fileStream = new FileStream(fullPath, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, useAsync: false);
+                    using (XmlReader xr = XmlReader.Create(fileStream, xmlReaderSettings))
```
Comment (AndyGerlicher) [inline src/XMakeBuildEngine/Construction/ProjectRootElement.cs]: Did you want to comment how horrible this is? :cry:

Comment (AndyGerlicher) [inline src/XMakeBuildEngine/Construction/ProjectRootElement.cs]: Never mind, I see it in the commit message.

**Comment by AndyGerlicher** [comment]
LGTM except for the CI

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Pull request merged**